File: Code/groceries.R
## Market basket analysis at the grocery outlet ##

# Loading the package #
library(arules)

# Importing data #
groceries = read.csv("https://raw.githubusercontent.com/iese-bad/DataSci/master/Data/groceries.csv")
dim(groceries)

# Coerce to a matrix #
groceries = as.matrix(groceries)

# Mining itemsets #
item1 = apriori(groceries, parameter=list(support=0.01, minlen=1, maxlen=1,
  target="frequent itemsets"))
item1 = sort(item1, by='support')[1:20]
inspect(item1)
item2 = apriori(groceries, parameter=list(support=0.01, minlen=2, maxlen=2,
  target='frequent itemsets'))
item2 = sort(item2, by="support")[1:20]
inspect(item2)
item3 = apriori(groceries, parameter=list(support=0.01, minlen=3, maxlen=3,
  target='frequent itemsets'))
item3 = sort(item3, by="support")[1:20]
inspect(item3)

# Mining rules #
rules = apriori(groceries, parameter=list(support=0.01, confidence = 0.25,
  maxlen=2, minlen=2, target='rules'))
rules = sort(rules, by='confidence')[1:20]
inspect(rules)

File: Code/groceries.py
## Market basket analysis at the grocery outlet ##

# Importing data #
import pandas as pd
groceries = pd.read_csv("https://raw.githubusercontent.com/iese-bad/DataSci/master/Data/groceries.csv")
groceries.shape

# Mining itemsets #
from mlxtend.frequent_patterns import apriori
frequent_itemsets = apriori(groceries, min_support=0.01, use_colnames=True)
frequent_itemsets.info()
frequent_itemsets.head()
frequent_itemsets['length'] = frequent_itemsets['itemsets'].apply(lambda x: len(x))
frequent_itemsets.head()
item1 = frequent_itemsets[frequent_itemsets['length'] == 1]
item1.sort_values('support', ascending=0)[:20]
item2 = frequent_itemsets[frequent_itemsets['length'] == 2]
item2.sort_values('support', ascending=0)[:20]
item3 = frequent_itemsets[frequent_itemsets['length'] == 3]
item3.sort_values('support', ascending=0)[:20]

# Mining rules #
from mlxtend.frequent_patterns import association_rules
rules = association_rules(frequent_itemsets, metric="confidence", min_threshold=0.25)
rules = rules.sort_values('confidence', ascending=0)[0:20]
rules[['antecedents', 'consequents', 'support', 'confidence', 'lift']]
